trieste/acquisition/function.py

```diff
@@ -625,3 +666,59 @@ def sample(self, at: TensorType, *, jitter: float = DEFAULTS.JITTER) -> tf.Tenso
         new_order = tf.concat([leading_indices, absolute_trailing_indices], axis=0)
 
         return mean[..., None, :, :] + tf.transpose(variance_contribution, new_order)
+
+
+class BatchMonteCarloExpectedImprovement(SingleModelBatchAcquisitionBuilder):
+    """
+    Expected improvement for batches of points, approximated using Monte Carlo estimation with the
+    reparametrization trick. See :cite:`Ginsbourger2010` for details.
+    """
+
+    def __init__(self, sample_size: int, *, jitter: float = DEFAULTS.JITTER):
+        """
+        :param sample_size: The number of samples for each batch of points.
+        :param jitter: The size of the jitter to use when stabilising the Cholesky decomposition of
+            the covariance matrix.
+        :raise ValueError (or InvalidArgumentError): If ``sample_size`` is not positive, or
+            ``jitter`` is negative.
+        """
+        tf.debugging.assert_positive(sample_size)
+        tf.debugging.assert_greater_equal(jitter, 0.0)
+
+        super().__init__()
+
+        self._sample_size = sample_size
+        self._jitter = jitter
+
+    def __repr__(self) -> str:
+        """"""
+        return f"BatchMonteCarloExpectedImprovement({self._sample_size!r}, jitter={self._jitter!r})"
+
+    def prepare_acquisition_function(
+        self, dataset: Dataset, model: ProbabilisticModel
+    ) -> AcquisitionFunction:
+        """
+        :param dataset: The data from the observer. Must be populated.
+        :param model: The model over the specified ``dataset``. Must have event shape [1].
+        :return: The batch *expected improvement* acquisition function.
+        :raise ValueError (or InvalidArgumentError): If ``dataset`` is not populated, or ``model``
+            does not have an event shape of [1].
+        """
+        tf.debugging.assert_positive(len(dataset))
+
+        mean, _ = model.predict(dataset.query_points)
+
+        tf.debugging.assert_shapes(
+            [(mean, ["_", 1])], message="Expected model with event shape [1]."
+        )
+
+        eta = tf.reduce_min(mean, axis=0)
+        sampler = BatchReparametrizationSampler(self._sample_size, model)
+
+        def batch_ei(at: TensorType) -> TensorType:
+            samples = tf.squeeze(sampler.sample(at, jitter=self._jitter), axis=-1)  # [..., S, B]
+            min_sample_per_batch = tf.reduce_min(samples, axis=-1)  # [..., S]
+            batch_improvement = tf.maximum(eta - min_sample_per_batch, 0.0)  # [..., S]
+            return tf.reduce_mean(batch_improvement, axis=-1, keepdims=True)  # [..., 1]
```

Comment:
you might want to double check this, as I've reordered it to be faster

Reply:
LGTM